app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -381,10 +382,11 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int pos
                             .setOnClickListener(new View.OnClickListener() {
                                 @Override
                                 public void onClick(View view) {
-                                    Intent i = new Intent(Intent.ACTION_VIEW);
+                                    Intent i = new Intent(zulipApp.getApplicationContext(), PhotoViewActivity.class);
                                     i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                                    i.setData(Uri.parse(url));
+                                    i.putExtra(Intent.EXTRA_TEXT , url);
                                     zulipApp.startActivity(i);
+                                    ((Activity) context).overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
```

Comment:
I think this Transition will be perfect for this purpose.
![lydorea](https://cloud.githubusercontent.com/assets/18511177/21816666/f4653656-d786-11e6-8528-c1e1ca40e6d3.png)
![1cjqssa](https://cloud.githubusercontent.com/assets/18511177/21816670/f730794a-d786-11e6-8b31-3da9f7de9909.gif)
![rukkfuz](https://cloud.githubusercontent.com/assets/18511177/21816673/f8af8dc4-d786-11e6-80c7-9dfa23992dcc.gif)

https://guides.codepath.com/android/Shared-Element-Activity-Transition